HPCC-16226 WsDESDLConfig methods should echo resulting binding/def

rpastrana opened the pull request:
- Provides option to echo binding upon publish, or configuration change
- Provides option to echo defintion upon publish

Signed-off-by: rpastrana rodrigo.pastrana@lexisnexis.com

rpastrana committed:
HPCC-16226 Echo resulting binding

- Provides option to echo binding upon publish, or configuration change
- Provides option to echo defintion upon publish

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented on esp/scm/ws_esdlconfig.ecm:
```diff
@@ -25,20 +25,42 @@ ESPStruct MethodConfig
 {
     ESParray<ESPStruct MethodConfig, Method> Methods;
 };
+ESPstruct ESDLBinding
```
pre-existing, moved up in file for visibility

rpastrana commented on esp/scm/ws_esdlconfig.ecm:
```diff
@@ -25,20 +25,42 @@ ESPStruct MethodConfig
+    String Name;
+    int Seq;
+    String Id;
+    [min_ver("1.1")] String Interface;
```
pre-existing, moved up in file for visibility

rpastrana commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -217,6 +217,32 @@ bool ensureSDSPath(const char * sdsPath)
     return true;
 }
 
+void fetchESDLDefinitionFromDaliById(const char *id, StringBuffer & def)
```
pre-existing, moved up in file for visibility

rpastrana commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -476,20 +502,102 @@ bool CWsESDLConfigEx::onPublishESDLDefinition(IEspContext &context, IEspPublishE
+
+        msg.appendf("Successfully published %s", newqueryid.str());
+
+        double ver = context.getClientVersion();
```
after publishing the definition, fetch it from Dali and report it in response

rpastrana commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -955,6 +1170,112 @@ bool CWsESDLConfigEx::onConfigureESDLBindingMethod(IEspContext &context, IEspCon
                     publishESDLMethod(espProcName.str(), espBindingName.str(), esdlDefIdSTR.toLowerCase().str(), methodName, &item, override, msg);
                     resp.updateStatus().setDescription(msg.str());
+
+                    double ver = context.getClientVersion();
```
after configuring a method within a binding (ESDLBinding/Definition/Methods/Method) fetch the entire binding, and report it in the response

rpastrana commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1040,32 +1361,6 @@ int CWsESDLConfigEx::getBindingXML(const char * espProcName, const char * espBin
     }
 }
 
-void fetchESDLDefinitionFromDaliById(const char *id, StringBuffer & def)
```
moved up for visibility

rpastrana commented:
@wangkx please review.

HPCCSmoketest commented:
Automated Smoketest 
Sha: ea9465fdee3bb94abd435790ffd5ea5c850eac0d
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented on esp/scm/ws_esdlconfig.ecm:
```diff
@@ -25,20 +25,42 @@ ESPStruct MethodConfig
 {
     String Name;
     ESParray<ESPstruct NamedValue, Attribute> Attributes;
+    String Elements;
```
versioning?

rpastrana replied:
MethodConfig was added in 1.2, and all uses of the structure are versioned, so I think we're fine.

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -476,20 +502,102 @@ bool CWsESDLConfigEx::onPublishESDLDefinition(IEspContext &context, IEspPublishE
         }
-        else
+
+        msg.appendf("Successfully published %s", newqueryid.str());
```
In line 516, msg says "Could not publish...". But, "Successfully published..." appended here?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -727,6 +835,113 @@ bool CWsESDLConfigEx::onPublishESDLBinding(IEspContext &context, IEspPublishESDL
+                        IArrayOf<IEspMethodConfig> iesmethods;
+
+                        Owned<IPropertyTree> def;
+                        def.setown(esdlbindingtree->queryPropTree("Definition[1]"));
```
queryPropTree() should not setown?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -955,6 +1170,112 @@ bool CWsESDLConfigEx::onConfigureESDLBindingMethod(IEspContext &context, IEspCon
+                            IArrayOf<IEspMethodConfig> iesmethods;
+
+                            Owned<IPropertyTree> def;
+                            def.setown(esdlbindingtree->queryPropTree("Definition[1]"));
```
Same as line 849

wangkx commented:
@rpastrana looks good. A few comments.

rpastrana committed:
HPCC-16226 Code review changes

- Do not attempt to echo new changes, if changes failed
- Do no setown when querying  a tree

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@wangkx please review code review changes.

HPCCSmoketest commented:
Automated Smoketest 
Sha: ba6ac6dfd67dee40fdc412159b1e001335b485be
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented:
@rpastrana the changes look fine.

richardkchapman commented:
Again, commit title could be clearer

rpastrana commented:
@richardkchapman title updated, please merge.
